pkg/client/tokenservice/inmemory.go

```diff
@@ -21,7 +21,7 @@ import (
 )
 
 type inmemoryTokenService struct {
-	sync.Mutex
+	sync.RWMutex
```

Comment:
probably the syncing mechanism can be removed as the synchronization is already in-place in immuClient ?

Reply:
or isn't the tokenService deprecated?

Reply:
I removed mutex on immuclient